skyportal/handlers/api/followup_request.py

```diff
@@ -288,19 +357,48 @@ def post(self):
         """
         data = self.get_json()
         _ = Source.get_obj_if_owned_by(data["obj_id"], self.current_user)
-        data["start_date"] = arrow.get(data["start_date"]).datetime
-        data["end_date"] = arrow.get(data["end_date"]).datetime
         data["requester_id"] = self.associated_user_object.id
-        if isinstance(data["filters"], str):
-            data["filters"] = [data["filters"]]
-        followup_request = FollowupRequest(**data)
+        data['allocation_id'] = int(data['allocation_id'])
+
+        allocation = Allocation.query.get(data['allocation_id'])
+        if allocation is None:
+            return self.error('No such allocation.')
+        if allocation.group_id not in [
+            g.id for g in self.current_user.accessible_groups
+        ]:
+            return self.error('User does not have access to this allocation.')
+
+        instrument = allocation.instrument
+        if instrument.api_classname is None:
+            return self.error('Instrument has no remote API.')
+
+        if not instrument.api_class.implements_submit():
+            return self.error('Cannot submit followup requests to this Instrument.')
+
+        # validate the payload
+        jsonschema.validate(data['payload'], instrument.api_class.form_json_schema)
+
+        followup_request = FollowupRequest.__schema__().load(data)
         DBSession().add(followup_request)
         DBSession().commit()
 
         self.push_all(
             action="skyportal/REFRESH_SOURCE",
             payload={"obj_id": followup_request.obj_id},
         )
+
+        try:
+            instrument.api_class.submit(followup_request)
+        except Exception:
```

Comment:
Is a more specific exception raised that we can catch here, or is it really just a base `Exception` instance?

Reply:
all we are doing is adding logging of any exception raised during this call to the user on the frontend - after logging, the exception is still raised as normal